src/openrct2/object/Object.cpp

```diff
@@ -329,6 +329,48 @@ std::unique_ptr<IStream> ObjectAsset::GetStream() const
     return {};
 }
 
+u8string VersionString(const ObjectVersion& version)
+{
+    return std::to_string(std::get<0>(version)) + "." + std::to_string(std::get<1>(version)) + "."
+        + std::to_string(std::get<2>(version));
+}
+
+ObjectVersion VersionTuple(std::string_view version)
+{
+    if (version.empty())
+    {
+        return std::make_tuple(0, 0, 0);
+    }
+
+    auto nums = String::Split(version, ".");
+    uint16_t versions[VersionNumFields] = {};
+    if (nums.size() > VersionNumFields)
```

Comment:
Considering the warning says it expects X.Y.Z then this should perhaps be:
```suggestion
    if (nums.size() != VersionNumFields)
```
Which would also make the next check obsolete

Reply:
OpenRCT2's objects and most custom .parkobj have a version field of "1.0" so it is intended to support versions `X` `X.Y` `X.Y.Z`